packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -47,6 +47,65 @@ export function isAttributeValue(val: unknown): val is AttributeValue {
   return isValidPrimitiveAttributeValue(val);
 }
 
+export function truncateValueIfTooLong(
+  value: AttributeValue,
+  limit: number | undefined,
+  logger: Logger
+): AttributeValue {
+  if (limit == null) {
+    return value;
+  }
+
+  if (typeof value === 'boolean' || typeof value === 'number') {
+    // these types reasonably can't exceed a limit
+    // if the limit was set to e.g. 2, then true with a length of 4 will exceed a limit
+    // however, the purpose of the limit is to prevent accidents such as values being thousands of characters long
+    return value;
+  }
+
+  if (Array.isArray(value)) {
+    let accruedLength = 0;
+
+    // roughly measure the size of a serialized array
```

Comment:
if we add limit we should check the limit for all cases. The number might be 
`123131231231.321312312312` and if someone sets limit to 2 it should handle this case correctly too.

Reply:
Arrays truncation is now done using `JSON.stringify`. Truncation of numbers seems difficult to me in general. So I've addressed this in the spec by "attribute value size limit MUST NOT be set to any number lower than 32". The only number-like type which could exceed 32 chars when stringified is bigint, and it's not allowed as attribute value.